docs/architecture/adr-059-evidence-composition-and-lifecycle.md

```diff
@@ -0,0 +1,304 @@
+
+#### An example of a possible implementation path
+
+We would ignore amnesia evidence (as individually it's hard to make) and revert to the initial split we had before where `DuplicateVoteEvidence` is also used for light client equivocation attacks and thus we only need `LunaticEvidence`. We would also most likely need to remove `Verify` from the interface as this isn't really something that can be used.
+
+``` go
+type LunaticEvidence struct { // individual lunatic attack
+  header *Header
+  commonHeight int64
+  vote *Vote
+
+  timestamp time.Time // once again taken from the block time at the height of the common header
+}
+```
+
+### Hybrid Framework
+
+We allow individual and batch evidence to coexist together, meaning that verification, broadcasting and committing is done depending on the evidence type. A hybrid model effectively undermines any possibility of generalization and Tendermint's evidence module will use switch statements and concrete types instead of interfaces to correctly process the evidence. One could argue that although we could see more evidence on the horizon, this won't be to the extent of justifying generalizing evidence and that each evidence can make it's own unique way through the pipeline and on to being committed to the chain and broadcasted to the application.
+
+A hybrid framework (as well as a batch framework) also surfaces the problem of the interface between Tendermint and the application. Currently, we have one concrete type that captures the relevant information needed to communicate to the application. This concrete type is formed per validator and therefore makes it more difficult to assess cartel (multi-validator) attacks. This begs the question of whether Tendermint should, at the application level, break down the evidence for each of the validators or alter the ABCI data structure to include multiple validators that were involved at that height and thus better capture the full extent of the misbehavior.
+
+Last important note is that we are closest to this implementation than any of the other two.
+
+#### An example of a possible implementation path
+
+We would have to change the interface as it's not possible for `LightClientAttackEvidence` to return a single `Address()` so instead we would have `Addresses()`. In addition the actual infringing validators aren't known in `LightClientAttackEvidence` until verification is done as we are taking the union between the validator set at the common height and at the conflicting height.
+
+```go
+type Evidence interface { //proposed
+  Height() int64                                     // height of the equivocation
```

Comment:
These comments all say "equivocation," but it's not necessarily actually equivocation, right?

Reply:
correct - I just plucked it out of the code but these will need to be updated